File: app/src/main/java/com/elvotra/clean/presentation/contract/PostsContract.java
package com.elvotra.clean.presentation.contract;

import com.elvotra.clean.presentation.model.PostViewItem;

import java.util.List;

public interface PostsContract {

    interface View extends BaseView<IPostsPresenter> {
        void showPostsList(List<PostViewItem> postViewItems);

        void showPostDetails(int postId);

        void showNoResults();

        boolean isActive();
    }

    interface IPostsPresenter extends BasePresenter {
        void loadPosts(boolean forceUpdate);

        void openPostDetails(int postId);
    }
}

File: app/src/main/java/com/elvotra/clean/presentation/contract/PostsContract.kt
package com.elvotra.clean.presentation.contract

import com.elvotra.clean.presentation.model.PostViewItem

interface PostsContract {

    interface View : BaseView<IPostsPresenter> {

        val isActive: Boolean
        fun showPostsList(postViewItems: List<PostViewItem>)

        fun showPostDetails(postId: Int)

        fun showNoResults()
    }

    interface IPostsPresenter : BasePresenter {
        fun loadPosts(forceUpdate: Boolean)

        fun openPostDetails(postId: Int)
    }
}
